src/main/java/org/opentripplanner/client/parameters/TripPlanParameters.java

```diff
@@ -14,6 +14,7 @@ public record TripPlanParameters(
     int numItineraries,
     Set<RequestMode> modes,
     SearchDirection searchDirection,
+    Long searchWindow,
```

Comment:
This should be of type `java.time.Duration`.

Reply:
I know that Intellij complains when you pass an Optional as a constructor param and I don't agree with it.

Reply:
Or we can live the property as `Duration` and define a getter `Optional<Duration> getSearchWindow()`.

Reply:
I think records don't allow you to change the return type.  So we would have to convert to a class, which I'm fine with because the code has reached a certain complexity.

Reply:
Can you do the conversion? Intellij can do it for you.

Reply:
Records have getters without "get" preffix so this would be extra method. I will provide the patch and if not liked, I can convert record to class.

Reply:
I don't want to have two getters, so please do the conversion.